### ISSUE TITLE
put item with a non-key ConditionExpression - 'where' param being ignored

### ISSUE THREAD

**Issue by jpmtrabbold** [sensedeep/dynamodb-onetable#312]
First of all - thanks for this amazing library!

**Describe the bug**

It's not possible to use a ConditionExpression from the `where` param when putting an item using model.create, even though DynamoDB supports it.

**To Reproduce**

By using `someModel.create(keys, { exists: null, where: whereExpression })`, you'll see that the `whereExpression` is being ignored. **I'm really sorry** that I'm not providing a test case, but it's just that the code is **very** explicit in ignoring the `where` param when the operation is different than `'delete'` or `'update'`. Please consider the following function `addConditions` inside the file `Expression.js` - what I just described is on the last `if`:

```javascript
    /*
        Conditions for create | delete | update
        May also be used by 'get' in fallback mode.
     */
    addConditions(op) {
        let {conditions, params} = this
        let {hash, sort} = this.index
        if (params.exists === true) {
            conditions.push(`attribute_exists(#_${this.addName(hash)})`)
            if (sort) {
                conditions.push(`attribute_exists(#_${this.addName(sort)})`)
            }

        } else if (params.exists === false) {
            conditions.push(`attribute_not_exists(#_${this.addName(hash)})`)
            if (sort) {
                conditions.push(`attribute_not_exists(#_${this.addName(sort)})`)
            }
        }
        if (params.type && sort) {
            conditions.push(`attribute_type(#_${this.addName(sort)}, ${params.type})`)
        }
        if (op == 'update') {
            this.addUpdates()
        }
        // THIS IS THE CONDITION THAT CAUSES THE UNDESIRED BEHAVIOUR
        if (params.where && (op == 'delete' || op == 'update')) {
            conditions.push(this.expand(params.where))
        }
    }
```

**Expected behavior**
`whereExpression` should be applied to the conditions in the same way they are when using a `'delete'` or `'update'` operation. It could be as easy as changing this:

```javascript
        if (params.where && (op == 'delete' || op == 'update')) {
            conditions.push(this.expand(params.where))
        }
````

to this:
```javascript
        if (params.where) {
            conditions.push(this.expand(params.where))
        }
```

TIA

**Comment by mobsense** [comment]
Thanks for the feedback and report.

**Comment by mobsense** [comment]
Fixed in 2.3.5
